cmake/Modules/MacroQbtCommonConfig.cmake

```diff
@@ -0,0 +1,99 @@
+        INTERFACE
+            QBT_VERSION_MAJOR=${PROJECT_VERSION_MAJOR}
+            QBT_VERSION_MINOR=${PROJECT_VERSION_MINOR}
+            QBT_VERSION_BUGFIX=${PROJECT_VERSION_PATCH}
+            QBT_VERSION_BUILD=${PROJECT_VERSION_TWEAK}
+    )
+
+    set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.${PROJECT_VERSION_TWEAK}${QBT_VER_STATUS}")
+
+    add_library(qbt_common_cfg INTERFACE)
+
+    # Full C++ 14 support is required
+    target_compile_features(qbt_common_cfg
+        INTERFACE
+            cxx_std_14
+            cxx_contextual_conversions
+            cxx_binary_literals
+            cxx_decltype_auto
+            cxx_lambda_init_captures
+            cxx_generic_lambdas
+            cxx_variable_templates
+            cxx_relaxed_constexpr
+            cxx_attribute_deprecated
+            cxx_digit_separators
+    )
+
+    target_compile_definitions(qbt_common_cfg
+        INTERFACE
+            QBT_VERSION="v${PROJECT_VERSION}"
+            QBT_VERSION_2="${PROJECT_VERSION}"
+            QT_DEPRECATED_WARNINGS
+            QT_NO_CAST_TO_ASCII
+            QT_NO_CAST_FROM_BYTEARRAY
+            QT_USE_QSTRINGBUILDER
+            QT_STRICT_ITERATORS
+            $<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG_OUTPUT>
+            $<$<AND:$<BOOL:${UNIX}>,$<NOT:$<BOOL:${APPLE}>>,$<NOT:$<BOOL:${CYGWIN}>>>:_DEFAULT_SOURCE>
+            $<$<PLATFORM_ID:Windows>:
+                NTDDI_VERSION=0x06010000
+                _WIN32_WINNT=0x0601
+                _WIN32_IE=0x0601
+                UNICODE
+                _UNICODE
+                WIN32_LEAN_AND_MEAN
+                _CRT_SECURE_NO_DEPRECATE
+                _SCL_SECURE_NO_DEPRECATE
+                NOMINMAX>
+    )
+
+    # Clang 11 still doesn't support -Wstrict-null-sentinel
+    include(CheckCXXCompilerFlag)
+    check_cxx_compiler_flag(-Wstrict-null-sentinel SNS_SUPPORT)
+    target_compile_options(qbt_common_cfg
+        INTERFACE
+            $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:
+                -Wall
+                -Wextra
+                -Wcast-qual
+                -Wcast-align
+                -Winvalid-pch
+                -Woverloaded-virtual
+                -Wold-style-cast
+                -Wnon-virtual-dtor
+                -pedantic
+                -pedantic-errors>
+            $<$<AND:$<CXX_COMPILER_ID:GNU,Clang,AppleClang>,$<BOOL:${SNS_SUPPORT}>>:-Wstrict-null-sentinel>
+    )
+
+    target_link_options(qbt_common_cfg
+        INTERFACE
+            $<$<AND:$<BOOL:${MINGW}>,$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>>:--dynamicbase>
+    )
+
+    # Ignored on non-MSVC
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<BOOL:${MSVC_RUNTIME_DYNAMIC}>:DLL>")
+
+    # There is no generator expression for CMAKE_GENERATOR yet
+    if (CMAKE_GENERATOR MATCHES "Visual Studio")
+        target_compile_options(qbt_common_cfg
+            INTERFACE
+                /MP
```

Comment:
It is overkill to split it into several lines, it isn't that long:
```cmake
target_compile_options(qbt_common_cfg INTERFACE /MP)
```

Reply:
The point was to make it consistent with all the other such statements. But sure, I think the case where there's only one element can be the exception.

Reply:
All such instances fixed.